### ISSUE TITLE
Investigate and fix instabilities in messaging tests

### ISSUE THREAD

**Issue by mjurc** [quarkus-qe/quarkus-openshift-test-suite#93]
As of Quarkus 1.7, the messaging tests have been consistently failing in both `surefire` and `failsafe` stage.

The testing deployments and the test assertions need to be reviewed.

**Comment by Sgitario** [comment]
I've been trying to investigate the instabilities in the messaging tests and I found a couple of issues:

- In amqp-reactive without running on Openshift: the AbstractAMQPTest is expecting a list of prices [10 20 ... 100], because the producer sends a price every 1 second. The problem is when the producer emits a new tick after the 10th price before the test checks the result, it will produce another 10 value and the test will not work as the list of prices will be [10 20 ... 100 10]. 

This can be easily fixed by either limiting the producer to emit only 10 values or change the expectation to check whether the result contains 10 20 ... 100 (not equal).

- In amqp-reactive again, but running with Openshift, it seems that the application randomly hangs to connect with the AMQ running in the namespace. I could see this exception:

```
WARNING [io.ver.cor.imp.BlockedThreadChecker] (vertx-blocked-thread-checker) Thread Thread[vert.x-eventloop-thread-0,5,main]=Thread[vert.x-eventloop-thread-0,5,main] has been blocked for 3764 ms, time limit is 2000 ms: io.vertx.core.VertxException: Thread blocked
	at java.base@11.0.8/java.net.Inet6AddressImpl.lookupAllHostAddr(Native Method)
	at java.base@11.0.8/java.net.InetAddress$PlatformNameService.lookupAllHostAddr(InetAddress.java:929)
	at java.base@11.0.8/java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1515)
	at java.base@11.0.8/java.net.InetAddress$NameServiceAddresses.get(InetAddress.java:848)
```

I need further investigation about how to workaround this issue.

My question now is whether you experienced other or the same issues (as I only spotted flaky tests in the same project "amqp-reactive"). If so, can we see the traces somewhere?

**Comment by rsvoboda** [comment]
@mjurc will ping you and provide link to the job where you can experiment with custom branches etc.
We do not keep history of these fails, only last 10 runs.

**Comment by Sgitario** [comment]
> I've been trying to investigate the instabilities in the messaging tests and I found a couple of issues:
> 
> * **In amqp-reactive without running on Openshift**: the AbstractAMQPTest is expecting a list of prices [10 20 ... 100], because the producer sends a price every 1 second. The problem is when the producer emits a new tick after the 10th price before the test checks the result. At this scenario, the test will not work as the list of prices will be [10 20 ... 100 10].
> 
> This can be easily fixed by either limiting the producer to emit only 10 values or change the expectation to check whether the result contains 10 20 ... 100 (not equal).
> 
> * **In amqp-reactive again, but running with Openshift**: it seems that the application randomly hangs to connect with the AMQ instance running in the namespace. I could see this exception in the quarkus app:
> 
> ```
> WARNING [io.ver.cor.imp.BlockedThreadChecker] (vertx-blocked-thread-checker) Thread Thread[vert.x-eventloop-thread-0,5,main]=Thread[vert.x-eventloop-thread-0,5,main] has been blocked for 3764 ms, time limit is 2000 ms: io.vertx.core.VertxException: Thread blocked
> 	at java.base@11.0.8/java.net.Inet6AddressImpl.lookupAllHostAddr(Native Method)
> 	at java.base@11.0.8/java.net.InetAddress$PlatformNameService.lookupAllHostAddr(InetAddress.java:929)
> 	at java.base@11.0.8/java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1515)
> 	at java.base@11.0.8/java.net.InetAddress$NameServiceAddresses.get(InetAddress.java:848)
> ```
> 
> I need further investigation about how to workaround this issue.
> 
> My question now is whether you experienced the same issues or different ones (as I only spotted flaky tests in the same project "amqp-reactive"). If so, can we see the traces somewhere?
> 
> cc @rsvoboda @mjurc

Update about the second issue. This seems to be caused by some wrong settings:

- In [broker.xml](https://github.com/quarkus-qe/quarkus-openshift-test-suite/blob/0aca538a62df35744dda98faff59cf70db538195/messaging/amqp-reactive/src/test/resources/broker.xml#L9):

```
<acceptors>
    <acceptor name="activemq">tcp://localhost:61616</acceptor>
    <acceptor name="amqp">tcp://localhost:5672?protocols=AMQP</acceptor>
</acceptors>
```

- And in the [application.properties](https://github.com/quarkus-qe/quarkus-openshift-test-suite/blob/0aca538a62df35744dda98faff59cf70db538195/messaging/amqp-reactive/src/main/resources/application.properties#L8): 

```
# Configures the AMQP broker credentials.
amqp-username=quarkus
amqp-password=quarkus
%test.amqp-host=localhost
amqp-host=amq-broker-amqp
amqp-port=5672
```

The app sometimes uses the endpoint pointing to "localhost" and hence not working in Openshift. Updating the broker file to listen to all the network interfaces makes the tests consistently worked:

```
<acceptors>
    <acceptor name="activemq">tcp://0.0.0.0:61616</acceptor>
    <acceptor name="amqp">tcp://0.0.0.0:5672?protocols=AMQP</acceptor>
</acceptors>
```

Note that the security is disabled, so the properties amqp-username and amqp-password are not needed.

What do you think?

**Comment by rsvoboda** [comment]
@Sgitario my position is that you can propose whatever changes you like / you think are valid.
On the first sight all your comments look valid and reasonable.

Feel free to prepare PR with proposed changes and we will see how it goes.

**Comment by Sgitario** [comment]
The PR https://github.com/quarkus-qe/quarkus-openshift-test-suite/pull/106 has been merged, so I think this issue can be marked as resolved.

**Comment by rsvoboda** [comment]
yup, thanks for reminder

good practice is to write "Fixes #93" as part of the PR description. This way the issues gets automatically closed once the PR is merged.
